slippers/templatetags/slippers.py

```diff
@@ -75,16 +77,20 @@ def extract_template_parts(code: str) -> Tuple[str, str]:
     return "", code
 
 
-def resolve_value(value, context):
+def resolve_value(key, value, context, tag_name):
     """Resolve a value
 
     If value references context, resolve it
     If value is a stringified dict or list, convert it to a real dict or list
     """
     resolved = value.resolve(context)
-    if isinstance(resolved, str):
-        if resolved and ((resolved[0] == '{' and resolved[-1] == '}') or (resolved[0] == '[' and resolved[-1] == ']')):
-            return json.loads(resolved)
+
+    try:
+        if isinstance(resolved, str):
+            if resolved and ((resolved[0] == '{' and resolved[-1] == '}') or (resolved[0] == '[' and resolved[-1] == ']')):
+                return json.loads(resolved)
+    except JSONDecodeError as exception:
+        raise SlippersError(msg=f"Failed to parse JSON", argument=key, reraised=exception.msg, component=tag_name)
```

Comment:
Check out [exception chaining](https://docs.python.org/3/tutorial/errors.html#exception-chaining). I think that would be better than passing in the original exception's message.

Reply:
good idea https://github.com/QuickRelease/slippers/pull/25/commits/216725309f37a0457e66b2c76f8df5900dc34183